components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/MetaDataProviderFactory.scala

```diff
@@ -1,32 +1,29 @@
 package pl.touk.nussknacker.sql.db.schema
 
-import com.zaxxer.hikari.util.DriverDataSource
 import pl.touk.nussknacker.engine.util.ThreadUtils
+import pl.touk.nussknacker.sql.DbEnricherName
 import pl.touk.nussknacker.sql.db.ignite.IgniteMetaDataProvider
-import pl.touk.nussknacker.sql.db.pool.DBPoolConfig
-import pl.touk.nussknacker.sql.db.schema.MetaDataProviderFactory.igniteDriverPrefix
+import pl.touk.nussknacker.sql.db.pool.{DBPoolConfig, HikariDataSourceFactory}
 
 import java.sql.Connection
-import java.util.Properties
 
 object MetaDataProviderFactory {
   private val igniteDriverPrefix = "org.apache.ignite.IgniteJdbc"
-}
-
-class MetaDataProviderFactory {
 
-  def create(dbPoolConfig: DBPoolConfig): JdbcMetaDataProvider = {
-    val props = new Properties()
-    dbPoolConfig.dataSourceProperties.foreach { case (k, v) =>
-      props.put(k, v)
-    }
+  def create(dbPoolConfig: DBPoolConfig, dbEnricherName: DbEnricherName): JdbcMetaDataProvider = {
     val ds = ThreadUtils.withContextClassLoader(getClass.getClassLoader) {
-      new DriverDataSource(
-        dbPoolConfig.url,
-        dbPoolConfig.driverClassName,
-        props,
-        dbPoolConfig.username,
-        dbPoolConfig.password
+      HikariDataSourceFactory(
+        dbPoolConfig.copy(
+          // We always keep one idle connection to decrease the time needed to compile a node
+          initialSize = 1,
+          // We limit the pool to 1 connection instead of using original dbPoolConfig.maxTotal because we
+          // have two separate pools (one for metadata and one for runtime) and we want to reduce the risk
+          // that we exceed the limit of connections
+          maxTotal = 1,
+          // We don't want to monitor this pool because it is not used operationally
```

Comment:
But we may be interested in its activity on UI, and this monitoring is basically free?